gridformat/vtk/vtu_writer.hpp

```diff
@@ -56,6 +56,16 @@ class VTUWriter : public VTK::XMLWriterBase<Grid, VTUWriter<Grid>> {
             this->_set_data_array(context, "Piece.CellData", name, vtk_cell_fields.get(name));
         });
 
+        // set default active arrays (scalars, vectors, tensors)
+        for (std::size_t i = 1; i <= 3; ++i)
+        {
+            if (const auto cell_default = this->first_cell_field(i); !cell_default.empty())
+                this->_set_attribute(context, "Piece.CellData", VTK::active_array_attribute[i-1], cell_default);
+
+            if (const auto point_default = this->first_point_field(i); !point_default.empty())
+                this->_set_attribute(context, "Piece.PointData", VTK::active_array_attribute[i-1], point_default);
+        }
+
```

Comment:
We could put this into the `VTK::XMLWriterBase` class, then we don't have to implement it in all the xml flavour implementations? We could do it in the initial xml setup function (`_get_write_context`) or in (`_write_xml`) maybe? I think the first one would be best

Reply:
I'll make a suggestion becuase I'll push the gcc warning fix, anyway